data_steward/utils/bq.py

```diff
@@ -240,3 +242,84 @@ def get_latest_validation_dataset_id(project_id):
             return sorted(validation_datasets, key=lambda x: x[0],
                           reverse=True)[0][1]
     return None
+
+
+def dataset_exists(dataset_id, project_id):
+    """
+    Checks if the dataset exists via dataset id
+
+    :param dataset_id:  ID (name) of the dataset to check
+    :param project_id:  ID (name) of the project to check for the dataset
+
+    :raises:  RuntimeError if a project_id is not supplied
+    :raises:  RuntimeError if a dataset_id is not supplied
+    :returns:  True if the dataset exists
+    :returns:  False if the dataset does not exist
+    """
+    if not project_id:
+        raise RuntimeError("Please specify a project")
+    if not dataset_id:
+        raise RuntimeError("Please specify a dataset to check if exists")
+    else:
+        client = get_client(project_id)
+        dataset_id = f"{project_id}.{dataset_id}"
+        dataset = bigquery.Dataset(dataset_id)
+
+        try:
+            client.get_dataset(dataset)
+            return True
+        except NotFound:
+            return False
+
+
+def create_dataset(dataset_id, description, label, tag, project_id):
+    """
+    Creates a new dataset
+
+    :param project_id:  ID (name) of the project in which to create the dataset, is required
+    :param dataset_id:  ID (name) to give the new dataset, is required
+    :param description:  description of the dataset, is required
+    :param label:  Dict[str,str] labels for the dataset
+    :param tag: Dict[str, ''] tag for the dataset
+
+    :raises:  RuntimeError if the dataset does not have project_id
+    :raises:  RuntimeError if the dataset does not have dataset_id
+    :raises:  RuntimeError if the dataset does not have a description
+    :raises:  RuntimeError if the dataset does not have a label or table
+
+    :returns:  A bigquery Client object
+    """
+    if dataset_exists(dataset_id, project_id) == False:
+        if description.isspace() or not description:
+            raise RuntimeError(
+                "Please provide a description to create a dataset")
+
+        if not label and not tag:
+            raise RuntimeError(
+                "Label and/or tag is required to create a dataset")
+
+        dataset_id = f"{project_id}.{dataset_id}"
+
+        # Construct a full Dataset object to send to the API.
+        dataset = bigquery.Dataset(dataset_id)
+        dataset.description = description
+
+        if label is None:
+            dataset.labels = tag
+        elif tag is None:
+            dataset.labels = label
+        else:
+            label_and_tag = {}
+
+            for item in [label, tag]:
+                label_and_tag.update(item)
```

Comment:
Conceptually, is there a difference in a label and a tag?  I thought a tag was a key with a blank value (empty string).

Reply:
Yes, it is so conceptually, there is no difference between the two. However, @ksdkalluri suggested to separate the two to check if either one or both are specified.

Reply:
This is how I had it before as a parameter called `label_or_tag`, @ksdkalluri will have to elaborate on his reasoning.

Reply:
@lrwb-aou I asked to make the distinction so that a developer passing the arguments(label/key) will get to read the distinction b/w both. If that's too lenient we can still make it one argument and have doc string specify the distinction.